bokehjs/src/coffee/models/glyphs/hex_tile.ts

```diff
@@ -0,0 +1,279 @@
+
+    const n = self._q.length
+
+    const size = this.model.size
+    const aspect_scale = this.model.aspect_scale
+
+    self._x = new Float64Array(n)
+    self._y = new Float64Array(n)
+
+    if (this.model.orientation == "pointytop") {
+      for (let i = 0; i < n; i++) {
+        self._x[i] = size * Math.sqrt(3) * (self._q[i] + self._r[i]/2) / aspect_scale
+        self._y[i] = -size * 3/2 * self._r[i]
+      }
+    }
+    else {
+      for (let i = 0; i < n; i++) {
+        self._x[i] = size * 3/2 * self._q[i]
+        self._y[i] = -size * Math.sqrt(3) * (self._r[i] + self._q[i]/2) * aspect_scale
+      }
+    }
+
+    self.scale = self._scale
+  }
+
+
+  protected _index_data(): SpatialIndex {
+    let ysize = this.model.size
+    let xsize = Math.sqrt(3)*ysize/2
+
+    if (this.model.orientation == "flattop") {
+      [xsize, ysize] = [ysize, xsize]
+      ysize *= this.model.aspect_scale
+    }
+    else {
+      xsize /= this.model.aspect_scale
+    }
+
+    const points = []
+    for (let i = 0; i < this._x.length; i++) {
+      const x = this._x[i]
+      const y = this._y[i]
+      if (isNaN(x+y) || !isFinite(x+y))
+        continue
+      points.push({minX: x-xsize, minY: y-ysize, maxX: x+xsize, maxY: y+ysize, i})
+    }
+    return new RBush(points)
+  }
+
+  // overriding map_data instead of _map_data because the default automatic mappings
+  // for other glyphs (with cartesian coordinates) is not useful
+  map_data(): void {
+    const self = this as any
+
+    [self.sx, self.sy] = this.map_to_screen(self._x, self._y);
+
+    [self.svx, self.svy] = this._get_unscaled_vertices()
+
+  }
+
+  protected _get_unscaled_vertices(): [number[], number[]] {
+    const size = this.model.size
+    const aspect_scale = this.model.aspect_scale
+
+    if (this.model.orientation == "pointytop") {
+      const rscale = this.renderer.yscale
+      const hscale = this.renderer.xscale
+
+      const r = Math.abs(rscale.compute(0) - rscale.compute(size))                               // assumes linear scale
+      const h = Math.sqrt(3)/2*Math.abs(hscale.compute(0) - hscale.compute(size)) / aspect_scale // assumes linear scale
+      const r2 = r/2.0
+
+      const svx = [0, -h,  -h,   0,  h,  h ]
+      const svy = [r,  r2, -r2, -r, -r2, r2]
+
+      return [svx, svy]
+    }
+
+    else {
+      const rscale = this.renderer.xscale
+      const hscale = this.renderer.yscale
+
+      const r = Math.abs(rscale.compute(0) - rscale.compute(size))                               // assumes linear scale
+      const h = Math.sqrt(3)/2*Math.abs(hscale.compute(0) - hscale.compute(size)) * aspect_scale // assumes linear scale
+      const r2 = r/2.0
+
+      const svx = [r,  r2, -r2, -r, -r2, r2]
+      const svy = [0, -h,  -h,   0,  h,  h ]
+
+      return [svx, svy]
+    }
+
+  }
+
+  protected _render(ctx: Context2d, indices: number[], {sx, sy, scale}: HexTileData): void {
+
+    const [svx, svy] = this._get_unscaled_vertices()
```

Comment:
This was already computed in `map_data()`.

Reply:
It's not available to hover, selection glyphs, and they don't work.

Reply:
So, please add this as a comment, because it's likely I will be tripping over this sooner than later.

Reply:
same thing with `scale` IIRC, for now it has to be passed in to `_render` (or computed in render) for alt glyphs to work. In the future, the `GlyphRenderer` should take responsibility for these computed quantities so that all glyphs can see it.

Reply:
I realized I could pass it to render by just putting it in the argument that unpacks the view attributes